modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java

```diff
@@ -54,25 +61,37 @@ private void _upgradeDDMTemplates() throws Exception {
 					Matcher templateMatcher = getTemplatePattern().matcher(
 						script);
 
-					if (templateMatcher.find()) {
-						script = templateMatcher.replaceAll(
+					while (templateMatcher.find()) {
+						script = StringUtil.replace(
+							script, templateMatcher.group(),
 							getTemplatePatternReplacement());
 
+						if (Validator.isNotNull(getTemplateContextVariable())) {
+							String matcherValue = templateMatcher.group();
+
+							String matcherVariableName = matcherValue.substring(
+								0, matcherValue.indexOf(StringPool.EQUAL));
+
+							script = StringUtil.replace(
+								script, matcherVariableName.trim(),
```

Comment:
@velosofelipe64, in order to optimize, can you please create a method to hold all the logic to get the variable name? This way, we can reuse here and in the method manipulating the fragments. :wink: 

I was thinking in something like:
```
script = StringUtil.replace(
	script, getVariableName(templateMatcher),
	getTemplateContextVariable());

...

private String getVariableName(Matcher matcher) {
	String matcherGroup = matcher.group();

	String variableName = matcherGroup.substring(
		0, matcherGroup.indexOf(StringPool.EQUAL));

	return variableName.trim();
}
```